src/shared/condition.c

```diff
@@ -1134,7 +1136,9 @@ static int condition_test_psi(Condition *c, char **env) {
                 return log_debug_errno(r, "Failed to parse loadavg: %s", c->parameter);
 
         r = read_resource_pressure(pressure_path, PRESSURE_TYPE_FULL, &pressure);
-        if (r == -ENODATA) /* cpu.pressure 'full' was added recently, fall back to 'some'. */
+        /* cpu.pressure 'full' is undefined at system level since 5.13,
+         * but remains reported and set to 0 for compatibility. Fall back to 'some' */
+        if (r == -ENODATA || (system_wide && streq(pressure_type, "cpu") && pressure.total == 0))
```

Comment:
Please check c->type instead of doing string comparison

Reply:
Thanks, issue addressed